framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -191,6 +191,76 @@ MeshDiagnosticsGenerator::checkSidesetsOrientation(const std::unique_ptr<MeshBas
                 " is consistently oriented with regards to the blocks it neighbors";
 
     diagnosticsLog(message, _check_sidesets_orientation, flipped_pairs.size());
+
+    // Now check that there is no sideset radically flipping from one side's normal to another
+    // side next to it, in the same sideset
+    // We'll consider pi / 2 to be the most steep angle we'll pass
+    unsigned int num_normals_flipping = 0;
+    Real steepest_side_angles = 1;
+    for (const auto & [elem_id, side_id, side_bid] : side_tuples)
+    {
+      if (side_bid != bid)
+        continue;
+      const auto & elem_ptr = mesh->elem_ptr(elem_id);
+
+      // Get side normal
+      const std::unique_ptr<const Elem> face = elem_ptr->build_side_ptr(side_id);
+      std::unique_ptr<FEBase> fe(FEBase::build(elem_ptr->dim(), FEType(elem_ptr->default_order())));
+      QGauss qface(elem_ptr->dim() - 1, CONSTANT);
+      fe->attach_quadrature_rule(&qface);
+      const auto & normals = fe->get_normals();
+      fe->reinit(elem_ptr, side_id);
+      mooseAssert(normals.size() == 1, "We expected only one normal here");
+      const auto & side_normal = normals[0];
+
+      // Compare to the sideset normals of neighbor sides in that sideset
+      for (const auto neighbor : elem_ptr->neighbor_ptr_range())
+        if (neighbor)
+          for (const auto neigh_side_index : neighbor->side_index_range())
+          {
+            // Check that the neighbor side is also in the sideset being examined
+            if (!boundary_info.has_boundary_id(neighbor, neigh_side_index, bid))
+              continue;
+
+            // We re-init everything for the neighbor in case it's a different dimension
+            std::unique_ptr<FEBase> fe_neighbor(
+                FEBase::build(neighbor->dim(), FEType(neighbor->default_order())));
+            QGauss qface(neighbor->dim() - 1, CONSTANT);
+            fe_neighbor->attach_quadrature_rule(&qface);
+            const auto & neigh_normals = fe_neighbor->get_normals();
+            fe_neighbor->reinit(neighbor, neigh_side_index);
```

Comment:
So I don't think we should need `fe_neighbor`, this shadowing `qface`, nor `neigh_normals` and we can just call `fe->reinit(neighbor, neigh_side_index)` and use `normals`

Reply:
If you take my advice, then on line 214, don't store the returned value as a reference (which I believe was my suggested change previously).

Reply:
I think it's also NBD to keep things as they are

Reply:
yeah this is over sidesets not meshes so there s already a pretty small multiplier on the number of times we do this